src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -155,6 +161,33 @@ public override bool TrySetPropertyValue(string name, object value)
             return true;
         }
 
+        /// <inheritdoc/>
+        public override bool TryAddNestedResource(string name, object deltaNestedResource)
+        {
+            if (name == null)
+            {
+                throw Error.ArgumentNull("name");
+            }
+
+            if (!_updatableProperties.Contains(name))
+            {
+                return false;
+            }
+
+            if (_deltaNestedResources.ContainsKey(name))
+            {
+                // Ignore duplicated nested resource.
+                return false;
+            }
+
+            // Add the nested resource in the hierarchy.
+            // Note: We shouldn't add the structural properties to the <code>_changedProperties</code>, which
+            // is used for keeping track of changed non-structural properties at current level.
+            _deltaNestedResources[name] = deltaNestedResource;
+
+            return true;
+        }
+
```

Comment:
I don't think we should add a new public api. 

TrySetPropertyValue is enough, because they have the same signature. You can move the logic into the "TrySetPropertyValue" just compare the "object" type.

or just make it private and call from TrySetPropertyValue

Reply:
If you think it's necessary, I think a corresponding method name "TryGetNestedResource(..)" is needed. 

TrySetPropertyValue vs TryGetPropertyValue

TryAddNestedResource vs TryGetNestedResource


---
In reply to: [190748771](https://github.com/OData/webapi/pull/1445#discussion_r190748771) [](ancestors = 190748771)

Reply:
Agreed. Changed as part of the consolidation related to SetProperty().
Now it is an internal method:
private bool TrySetNestedResourceInternal(string name, object deltaNestedResource)

---
In reply to: [190749752](https://github.com/OData/WebApi/pull/1445#discussion_r190749752) [](ancestors = 190749752,190748771)